plugins/weather.py

```diff
@@ -63,61 +74,83 @@ def find_location(location):
     request.raise_for_status()
 
     json = request.json()
-    error = check_status(json['status'])
-    if error:
-        raise APIError(error)
+
+    if json['status'] != 'OK':
+        raise GeocodeAPIError(json['status'])
 
     return json['results'][0]['geometry']['location']
 
 
 def load_cache(db):
-    global location_cache
-    location_cache = []
+    """
+    :type db: sqlalchemy.orm.Session
+    """
     for row in db.execute(table.select()):
         nick = row["nick"]
         location = row["loc"]
-        location_cache.append((nick, location))
+        location_cache[nick] = location
```

Comment:
Clear the cache before reloading it, as you could have old user data still there.